src/java/org/pantsbuild/tools/junit/impl/AntJunitXmlReportListener.java

```diff
@@ -44,6 +44,24 @@
  */
 class AntJunitXmlReportListener extends RunListener {
 
+  // NB: Cache the localhost name as a workaround for an issue where localhost resolution on OSX
+  // may take seconds.
+  // See: https://bugs.openjdk.java.net/browse/JDK-8170910
+  private static String localHostName = null;
```

Comment:
Is it worth eagerly initializing this, or are there cases where laziness will avoid actually needing to construct it?

Reply:
Good question. I think I'm going to adjust some things. Since this is only necessary when the listener is used, I think I can hoist this into the listener's constructor instead and pass the hostname from the listener to the inner class that uses it on initialization.